### PR TITLE
Generate Windows version info for all dlls and exe's

### PR DESCRIPTION
**Pull request by TheOneRing**
Fixes: #6554

### CODE REVIEW

**Comment by TheOneRing** [comment]
image

**Review by michaelstingl — APPROVED** [review]
Nice one! 👍

File: cmake/modules/version.rc.in [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    FILEOS VOS__WINDOWS32
+    FILETYPE @OC_RC_TYPE@
+    FILESUBTYPE VFT2_UNKNOWN
+    BEGIN
+        BLOCK "StringFileInfo"
+        BEGIN
+            BLOCK "040904E4"
+            BEGIN
+                VALUE "CompanyName", "@APPLICATION_VENDOR@\0"
+                VALUE "LegalCopyright", "Copyright (C) 2014-@MIRALL_VERSION_YEAR@ @APPLICATION_VENDOR@\0"
```
Comment (TheOneRing) [inline cmake/modules/version.rc.in:21]: The start date is from the shell integration .rc files.
Maybe we want to use 2011 or don't provide a start date.

File: cmake/modules/version.rc.in [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#endif
+    FILEOS VOS__WINDOWS32
+    FILETYPE @OC_RC_TYPE@
+    FILESUBTYPE VFT2_UNKNOWN
+    BEGIN
+        BLOCK "StringFileInfo"
+        BEGIN
+            BLOCK "040904E4"
+            BEGIN
+                VALUE "CompanyName", "@APPLICATION_VENDOR@\0"
```
Comment (TheOneRing) [inline cmake/modules/version.rc.in:20]: The company name is branded, as before, was that really intended?

File: cmake/modules/version.rc.in (hunk 1 above)
Comment (TheOneRing) [reply cmake/modules/version.rc.in:21]: @hodyroff

**Review by dschmidt — APPROVED** [review]
🤩

**Pull request merged**